Arrowgene.Ddon.GameServer/Handler/LobbyLobbyDataMsgHandler.cs

```diff
@@ -53,7 +54,10 @@ public override void Handle(GameClient client, StructurePacket<C2SLobbyLobbyData
             else
             {
                 // We are in one of the common areas where players can see eachother
-                foreach (GameClient otherClient in Server.ClientLookup.GetAll())
+		List<GameClient> otherClients = Server.ClientLookup.GetAll()
+                                                    .Where(x => x != null)
+                                                    .ToList();
+		foreach (GameClient otherClient in otherClients)
```

Comment:
I don't like that we're using LINQ to filter and build a list and then iterate over it to filter it further. I think you should either do it all with LINQ, and finally run otherClient.Send(res) in a .ForEach, or dont use LINQ and add the null check inside the foreach.

Reply:
agreed and fixed